### DISCUSSION TITLE
How to install Knowlg

### DISCUSSION THREAD

**Discussion by coolbung** [Sunbird-Knowlg/Community#78] [Documentation Issues]
Please share documentation and server requirement for installing Knowlg. 
Hope it can be installed on a single server :)

**Comment by pramodkvarma** [comment]
Had discussed with @vrayulu

**Reply by maheshkumargangula** [reply]
@coolbung we are documenting the installation process for Sunbird Knowlg. I will share the details soon in this thread.

**Reply by Jayaprakash8887** [reply]
@vinukumar-vs

**Reply by vinukumar-vs** [reply]
@coolbung sorry to say, we are still working on installation steps of knowlg BB independently. But we can deploy on Knowlg in a single machine of 8core & 32GB. Below are the infra details


Service | CPU | Memory
-- | -- | --
Kubernetes | 4core | 16gb
Jenkins | 2core | 8gb
Knowlg services/DB | 8core | 32gb
Learning service(VM) | 1core | 1gb

Kubernetes & Jenkins is part of common setup not only specific to Knowlg. 


Knowlg BB : microservices, DB's & modules 


  |   | Recomended for Scale |  
-- | -- | -- | --
  | Microservice / DB / Product | CPU | Memory (min)
Service | Content service | 1 core | 1GB
  | Taxonomy service | 1 core | 1GB
  | DIAL service | 1 core | 1GB
  | Search service | 1 core | 1GB
  | Knowlg platform jobs | 1 core | 1GB
Databases | Neo4j | 7 core | 16GB
  | Cassandra | 2 core | 8GB
  | Redis | 2 core | 4GB
  | Logstash | 1 core | 2GB
  | Elasticsearch | 8 core | 16GB
  | Kafka | 4 core | 8GB
Front-end Modules | Content editors |   |  
  | Content players |   |  
Security | Knowlg mw | 1 core | 1GB

**Comment by maheshkumargangula** [comment]
@coolbung - We have the documentation about how to setup taxonomy-service with a Kubernetes cluster in a local machine.
https://github.com/project-sunbird/knowledge-platform/blob/release-5.2.0/KNOWLG-SETUP.md

All the Sunbird Knowlg microservices  works independently like taxonomy-service. Please use the `taxonomy-service` setup documentation as reference to setup other microservices. 

Cc: @vrayulu @pallakartheekreddy

**Comment by jayesh9747** [comment]
@coolbung  can you provide the assessment -service documentation ?

**Reply by pallakartheekreddy** [reply]
@jayesh9747 The assessment service is part of the Inquiry Building block.  Please check the installation guide for assessment service 
https://inquiry.sunbird.org/use/developer-installation/question-and-question-set-service/installation

Kindly submit your query on the forum provided below
https://github.com/orgs/Sunbird-inQuiry/discussions

**Comment by jayesh9747** [comment]
@pallakartheekreddy  can you  guide that , which apis contain in this folder assessment api ? 
![image](https://github.com/Sunbird-Knowlg/Community/assets/123955234/39051fd5-f54e-4418-9227-41f8ef3b059b)

**Reply by pallakartheekreddy** [reply]
@jayesh9747 As I said previously, These APIs are moved to InQuiry Building block. Please refer below link for more details about APIs
https://inquiry.sunbird.org/learn/product-and-developer-guide/question-and-question-set-service

**Reply by jayesh9747** [reply]
![image](https://github.com/Sunbird-Knowlg/Community/assets/123955234/bb9ba5a1-fa22-436b-b5eb-58c939d25958)
@pramodkvarma , in this architecture, the inquiry service depends on the knowlg database. I want to write a script that migrates my entire questionnaire database from MongoDB to the knowlg database. For that, I require an API endpoint where I can post data, so that the entire process is taken care of by the knowlg building block.

**Reply by pallakartheekreddy** [reply]
Hi @jayesh9747  **knowlg Core** is a software component which is utilised by inQuiry api service (assessment-service). 
For your migration script, you need to understand the api's offered by assessment-service and use them because direct db migration is not recommended. So migration has to be done through api's only. 

If you still have questions, lets discuss further on inQuiry discussion thread created by you (https://github.com/orgs/Sunbird-inQuiry/discussions/108)

**Comment by jayesh9747** [comment]
Sure sir , thank you .

**Reply by pallakartheekreddy** [reply]
We will proceed to close this thread. If you have any further questions or require assistance in the future, please don't hesitate to start a new thread. Thank you.
